ext/rotoscope/rotoscope.c

```diff
@@ -183,13 +231,7 @@ static void event_hook(VALUE tpval, void *data) {
   rs_tracepoint_t trace = extract_full_tracevals(trace_arg, &trace_path);
```

Comment:
I now return early if it's a return from a block method that _does not_ correspond to the last caller:

https://github.com/Shopify/rotoscope/blob/7427cd9d6786636a24ccb89ac4c124996ed2eeeb/ext/rotoscope/rotoscope.c#L182-L192

This should catch the case where it's a block return from a blacklisted method, but we still correctly handle unblacklisted block returns since their callers will still be in the call stack.

Reply:
We also want to fix the filepath before the rejected_path condition to make sure that it doesn't skip the conditional block when the method should have been blacklisted.

Reply:
I don't think that will ever happen because `last_frame->tp` will have had to pass through the `rejected_path` condition itself, so if it's in the stack it's a non-blacklisted path. We only ever need to fix up the filepaths of return events, so calls should be reliable.

Reply:
Oh, it looks like we are relying on log_rotoscope_trace to filter when the stacks don't match so that we don't log the return event when the call was filtered.  I guess that is correct but relying on that makes the code harder to understand.